src/CommitmentsV2/SFA.DAS.CommitmentsV2.MessageHandlers/EventHandlers/RecordedAct1CompletionPaymentEventHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.Logging;
+using NServiceBus;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
+
+namespace SFA.DAS.CommitmentsV2.MessageHandlers.EventHandlers
+{
+    public class RecordedAct1CompletionPaymentEventHandler : IHandleMessages<RecordedAct1CompletionPaymentFakeEvent>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        private readonly ILogger<RecordedAct1CompletionPaymentEventHandler> _logger;
+
+        public RecordedAct1CompletionPaymentEventHandler(Lazy<ProviderCommitmentsDbContext> dbContext, ILogger<RecordedAct1CompletionPaymentEventHandler> logger)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+        }
+
+        public async Task Handle(RecordedAct1CompletionPaymentFakeEvent message, IMessageHandlerContext context)
+        {
+            try
+            {
+                if (message.ApprenticeshipId.HasValue)
+                {
+                    var apprentice = await _dbContext.Value.Apprenticeships.Include(x=>x.Cohort).SingleAsync(x => x.Id == message.ApprenticeshipId);
+
+                    switch (apprentice.Status)
+                    {
+                        case ApprenticeshipStatus.Live:
+                            apprentice.Complete(message.EventTime.DateTime);
```

Comment:
What about Paused?

Reply:
I'd assumed from the comments in the technical notes. That we should only process the CompletionEvent if it was Live. 

The domain will throw it if it's not live, Do we then allow the domain to throw exceptions for Paused (and stopped) statuses and deal with these as poisoned messages

Reply:
Yes, you're quite right. See my extra note below.